src/services/textChanges.ts

```diff
@@ -1166,6 +1180,34 @@ namespace ts.textChanges {
                     Debug.assertNever(gp);
             }
         }
+
+        function getRangeToDeleteJsDocTag(node: JSDocTag, sourceFile: SourceFile): TextRange | undefined {
+            const { text } = sourceFile;
+            const { parent } = node;
+            if (parent.kind === SyntaxKind.JSDocComment && parent.comment === undefined && parent.tags!.length === 1) {
+                return createTextRangeFromNode(parent, sourceFile);
+            }
+
+            let pos = node.getStart(sourceFile);
+            Debug.assert(text.charCodeAt(pos) === CharacterCodes.at);
+            pos--;
+            while (isWhiteSpaceSingleLine(text.charCodeAt(pos))) pos--;
+
+            // If the line doesn't begin with `*`, give up.
+            if (text.charCodeAt(pos) !== CharacterCodes.asterisk) return undefined;
+
+            let { end } = node;
+            // Tag end *might* be at the end of the line, but might be at the next tag.
+            // So back up if necessary, then proceed to beginning of next tag.
+            while (text.charCodeAt(end - 1) === CharacterCodes.space) end--;
+            while (isLineBreak(text.charCodeAt(end - 1))) end--;
+
+            while (!isLineBreak(text.charCodeAt(end))) end++;
+            while (isLineBreak(text.charCodeAt(end))) end++;
+            while (isWhiteSpaceSingleLine(text.charCodeAt(end))) end++;
+
+            return text.charCodeAt(end) === CharacterCodes.asterisk ? { pos, end } : undefined;

```

Comment:
Same here — the following line also has to start with an asterisk? Seems like `@` should also be a boundary.